### DISCUSSION TITLE
How to use a custom font in Codespaces

### DISCUSSION THREAD

**Discussion by ghost** [community/community#42740] [Codespaces]
### Select Topic Area

Question

### Body

The default font used in Codespaces seems to be quite normal. I've been using JetBrains mono for linting and for its good design. Is there a way that I can install the font on GitHub Codespaces and use it? Please help

**Comment by cmbrose** [comment]
👋 Thanks for reaching out @ameyagore - I was able to use the font in a codespace on vscode desktop by following [this article](https://vueschool.io/articles/vuejs-tutorials/how-to-install-jetbrains-mono-font-in-visual-studio-code/).

To use the font in browser I followed pretty much the same steps, but I found that it worked when installing under `~/.local/share/fonts`, it didn't seem to work when installing under the global `/usr/share/fonts` directory.

**Reply by ghost** [reply]
Thanks for the reply. I actually figured it out. In my local VS Code environment, I had written in the settings.json file to use the JetBrains Mono font, and it was synced with my Microsoft account. I just simply logged in to my Microsoft icon in the Codespaces to sync the settings.json file, and it actually worked out.

**Reply by unrealapex** [reply]
I tried this and it unfortunately did not work. Do you have any idea why this might be? I am trying to do this with a patched font.

**Reply by ghost** [reply]
Hi, thank you for posting your doubt. I would be happy to help you out. You can have a look at these steps.

1. Open your Codespace.
2. Open 'settings.json' file.
3. Paste the following in 'settings.json' `"editor.fontFamily": "JetBrains Mono",`
4. First check if it's working with JetBrains Mono, if it does, then modify the same with the font of your choice.

Please reach out if you need more help!

**Reply by unrealapex** [reply]
To clarify, I am trying to change the integrated terminal's font.

**Reply by 0xF5T9** [reply]
Did anyone really successfully install custom fonts on codespaces?
I installed the fonts in `/usr/share/fonts` successfully, but vscode still doesn't recognize the font and uses the default one. Restarting the codespaces doesn't work.
I try to install the font in `~/.local/share/fonts` as mentioned by cmbrose, but the path doesn't exist in my codespace.

**Reply by iamo0** [reply]
I have the same issue. I've installed fonts on remote machine and changed settings but I don't see any changes.

**Comment by OwaisAzad** [comment]


**Reply by feetstv** [reply]
This doesn't do anything.

**Reply by fatherofinvention** [reply]
Was this an AI generated response? Hmm…

**Reply by unrealapex** [reply]
Now that you mention it, this does sound like something a language model would spit out.

**Comment by akashkennedy** [comment]
### To Load custom font on your vs code 
First You need to install the font on your system
Next go to vsc setting and search for font 
you can choose the  name of your font inside  single quote and a backup font after that
```
'ubuntu', monospace
```

**Comment by jpawlowski** [comment]
Does this really work without installing a font locally on my machine, only inside the Codespace container? Would this really work in the browser simply on any machine opening the Codespace?

I've been trying things for many hours now but had no success so far.

There are also quite many uncertainties like:

- Do we use TrueType font in static or variable variant? Probably static, but nothing is mentioned anywhere.
- For web compatibility, can Codespaces really make use of the TrueType fonts, or do we need WOFF2 format? Again, static, or variable variant? Can there be a conflict when installing both WOFF2 and TrueType in parallel so that for web clients, they can use the WOFF2 format, and VS Code desktop can use TrueType format?
- Do we need to have Fontconfig package installed in the Codespace container? Do we need to update the font cache after copying the font files (`fc-cache -f -v`)?
- At what point in the Codespace container provisioning process should the font files be installed, should they already be part of a pre-build image maybe?

**Comment by cecilialee** [comment]
I'm using codespace on ipad. I tried the options above, installing the font in the container and both on my ipad device but not working.

At the end I installed VSCode extensions inside codespace - I tried two of them and both working
- Fira Code Nerd Font / Icons
- Font: JetBrainsMono Nerd Font

Now I'm happy - JetBrainsMono always my fav font!
